GetPet/GetPet.Crawler/CrawlerBase.cs

```diff
@@ -39,10 +40,21 @@ public virtual IList<PetDto> Parse()
 
         public virtual void InsertToDB(IPetHandler db, IList<PetDto> input)
         {
-            foreach (var pet in input)
+            var animals = input.Where(p => !IsPetExists(db, p));
+
+            foreach (var pet in animals)
             {
                 db.AddPet(pet);
             }
         }
+
+        private bool IsPetExists(IPetHandler db, PetDto pet)
+        {
+            var allAnimals = new List<PetDto>(); // TODO: Get all animals from DB
+
+            // TODO: Add source (which website) to animals
+
+            return (allAnimals.Any(p => p.Name == pet.Name)); // TODO: Better comapring conditions. Consider using 'Equals'
```

Comment:
name is not unique enough,
try to add a column to pet table - ExternalIdentifer & map it to the id of the crawled site,
eg. "Yad2:123132"